solver/src/settlement_submission.rs

```diff
@@ -29,9 +24,7 @@ pub async fn submit(
     gas_price_cap: f64,
     settlement: Settlement,
 ) -> Result<()> {
-    let settlement = settlement.into();
-    // Check that a simulation of the transaction works before submitting it.
-    simulate_settlement(&settlement, contract).await?;
+    let settlement: EncodedSettlement = settlement.into();
```

Comment:
Note that this means that this function `submit` no longer does any simulation because the driver simulates all settlements beforehand. I'd also be okay to simulate anyway for extra safety.